source/custom/xnetsocket_converters.h

```diff
@@ -282,6 +282,83 @@ inline SockOptDataInputConverter convert_from_grpc(const SockOptData& input)
   return SockOptDataInputConverter(input);
 }
 
+// This class allows us to have something allocated on the stack that provides backing
+// storage for a void* opt_val output param and converts it to the SockOptData grpc-type.
+struct SockOptDataOutputConverter {
+  SockOptDataOutputConverter(int32_t opt_name) : opt_name(opt_name)
+  {
+  }
+
+  void* data()
+  {
+    switch (opt_name) {
+      case OptName::OPT_NAME_IP_MULTICAST_TTL:
+      case OptName::OPT_NAME_IPV6_MULTICAST_HOPS:
+      case OptName::OPT_NAME_SO_RX_DATA:
+      case OptName::OPT_NAME_SO_RCV_BUF:
+      case OptName::OPT_NAME_SO_SND_BUF:
+      case OptName::OPT_NAME_TCP_NODELAY: {
+        return &data_int;
+        break;
+      }
+      case OptName::OPT_NAME_SO_LINGER:
+      case OptName::OPT_NAME_SO_NON_BLOCK:
+      case OptName::OPT_NAME_SO_REUSE_ADDR: {
+        return &data_bool;
+        break;
+      }
+      case OptName::OPT_NAME_SO_BIND_TO_DEVICE:
+      case OptName::OPT_NAME_SO_ERROR: {
+        data_string = std::string(256 - 1, '\0');  // TODO: What's the max string size to allocate for a sock opt?
```

Comment:
Need a task for this todo

Reply:
Made AB#1888861 to track this. Alex is waiting for these changes so I'll go ahead and submit this PR instead of adding to the comment and having waiting for the build again.